twilight-lavalink/src/model/incoming.rs

```diff
@@ -0,0 +1,363 @@
+    Ready,
+    /// Updated statistics about a node.
+    Stats,
+}
+
+use serde::{Deserialize, Serialize};
+use twilight_model::id::{marker::GuildMarker, Id};
+
+/// The levels of severity that an exception can have.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub enum Severity {
+    /// The cause is known and expected, indicates that there is nothing wrong with the library itself.
+    Common,
+    /// The probable cause is an issue with the library or there is no way to tell what the cause might be. This is the default level and other levels are used in cases where the thrower has more in-depth knowledge about the error.
+    Fault,
+    /// The cause might not be exactly known, but is possibly caused by outside factors. For example when an outside service responds in a format that we do not expect.
+    Suspicious,
+}
+
+/// The exception with the details attached on what happened when making a query to lavalink.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct Exception {
+    /// The cause of the exception.
+    pub cause: String,
+    /// The message of the exception.
+    pub message: Option<String>,
+    /// The severity of the exception.
+    pub severity: Severity,
+}
+
+/// An incoming event from a Lavalink node.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(untagged)]
+pub enum IncomingEvent {
+    /// Dispatched when player or voice events occur.
+    Event(Event),
+    /// Dispatched when you successfully connect to the Lavalink node.
+    Ready(Ready),
+    /// New statistics about a node and its host.
+    Stats(Stats),
+    /// An update about the information of a player.
+    PlayerUpdate(PlayerUpdate),
+}
+
+impl From<Ready> for IncomingEvent {
+    fn from(event: Ready) -> IncomingEvent {
+        Self::Ready(event)
+    }
+}
+
+impl From<Event> for IncomingEvent {
+    fn from(event: Event) -> IncomingEvent {
+        Self::Event(event)
+    }
+}
+
+impl From<PlayerUpdate> for IncomingEvent {
+    fn from(event: PlayerUpdate) -> IncomingEvent {
+        Self::PlayerUpdate(event)
+    }
+}
+
+impl From<Stats> for IncomingEvent {
+    fn from(event: Stats) -> IncomingEvent {
+        Self::Stats(event)
+    }
+}
+
+/// The discord voice information that lavalink uses for connection and sending information.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct VoiceState {
+    /// The Discord voice endpoint to connect to.
+    pub endpoint: String,
+    /// The Discord voice session id to authenticate with. Note this is separate from the lavalink session id.
+    pub session_id: String,
+    /// The Discord voice token to authenticate with.
+    pub token: String,
+}
+
+/// An update about the information of a player. Filters are currently unsupported
```

Comment:
This is oddly worded. Something like `An update for a player's state` probably carries more meaning. Also, I'm not a native speaker, but it should be `update of` or `update on` and not `update about` if I'm not mistaken